Tests/BusinessLogic.Tests/UnitTests/LogicTests/PlayedGamesTests/PlayedGameRetrieverTests/SearchPlayedGamesTests.cs

```diff
@@ -365,10 +367,37 @@ public void ItFiltersOnTheInclusionExternalSourceName()
             };
 
             //--act
-            var results = autoMocker.ClassUnderTest.SearchPlayedGames(filter);
+            var results = autoMocker.ClassUnderTest.SearchPlayedGames(filter, null);
 
             //--assert
             Assert.True(results.All(result => result.ApplicationLinkages.Any(x => x.ApplicationName == EXTERNAL_SOURCE_NAME)));
         }
+
+        [Test]
+        public void ItAddsPlayerIdToFilterIfWeOnlyWantCurrentPlayerGames()
+        {
+            ApplicationUser CurrentUser = new ApplicationUser()
+            {
+                Id = "1",
+                CurrentGamingGroupId = 1
+            };
+            Player CurrentPlayer = new Player()
+            {
+                Id = 1
+            };
+            PlayedGameFilter filter = new PlayedGameFilter()
+            {
+                IncludeCurrentPlayerGamesOnly = true
+            };
+            autoMocker.Get<IPlayerRetriever>().Expect(mock => mock.GetPlayerForCurrentUser(
+                    CurrentUser.Id, CurrentUser.CurrentGamingGroupId))
+                .Return(CurrentPlayer);
+
+            autoMocker = new RhinoAutoMocker<PlayedGameRetriever>();
+            autoMocker.Get<IDataContext>().Expect(mock => mock.GetQueryable<PlayedGame>()).Return(new List<PlayedGame>().AsQueryable());
+            autoMocker.ClassUnderTest.SearchPlayedGames(filter, null);
+
+            Assert.That(filter.PlayerId, Is.EqualTo(1));
+        }
```

Comment:
If we make the aforementioned change, I'd probably add a test for "ItThrowsAnArgumentNullExceptionIfTheApplicationUserIsNullButTheyAreTryingToFilterToOnlyTheCurrentUsersGames()"